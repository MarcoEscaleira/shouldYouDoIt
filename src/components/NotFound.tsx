import React from 'react';
import { Link } from 'react-router-dom';

const NotFound: React.FC = () => (
  <div className="notFound">
    <p className="notFound__label">404</p>
    <p className="notFound__text">
      Oops, we can't seem to find the page you're looking for.
    </p>
    <Link to="/" className="notFound__link">
      Go Home
    </Link>
  </div>
);

export default NotFound;
