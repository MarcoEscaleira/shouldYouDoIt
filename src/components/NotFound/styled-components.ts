// @ts-nocheck
import styled from "styled-components";
import { Link } from "react-router-dom";
import { pallete, sizes, normalFont } from "~/styles/_settings";

export const Container = styled.div`
  display: flex;
  width: 100%;
  padding: ${sizes.m};
  flex: 1;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  font-family: ${normalFont};
`;

export const Label = styled.p`
  font-size: 8rem;
  color: ${pallete.lightest};
`;

export const Text = styled.p`
  font-size: ${sizes.m};
  text-align: center;
  color: ${pallete.second_light};
  margin-bottom: ${sizes.xl};
`;

export const Button = styled(Link)`
  align-items: center;
  color: ${pallete.fifth_light};
  width: 10rem;
  height: 4rem;
  border-radius: 1rem;
  display: flex;
  justify-content: center;
  background-color: ${pallete.second_dark};
`;
