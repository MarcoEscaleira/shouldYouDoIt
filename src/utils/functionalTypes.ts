export type NullaryFn<R> = () => R;
export type UnaryFn<A, R> = (a: A) => R;
export type BinaryFn<A, B, R> = (a: A, b: B) => R;
export type TernaryFn<A, B, C, R> = (a: A, b: B, c: C) => R;
export type QuadFn<A, B, C, D, R> = (a: A, b: B, c: C, d: D) => R;
export type QuinFn<A, B, C, D, F, R> = (a: A, b: B, c: C, d: D, f: F) => R;
export type SixFn<A, B, C, D, F, H, R> = (
  a: A,
  b: B,
  c: C,
  d: D,
  f: F,
  h: H
) => R;
