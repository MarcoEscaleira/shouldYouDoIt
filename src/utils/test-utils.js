import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter as Router } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import store from '../store';

export const renderWithRedux = (ui, initialState) => {
  return {
    ...render(
      <Provider store={store(initialState)}>
        <Router>{ui}</Router>
      </Provider>
    )
  };
};

export const renderWithRouter = ui => {
  return {
    ...render(<Router>{ui}</Router>)
  };
};

const middlewares = [thunk];
export const mockStore = configureStore(middlewares);
