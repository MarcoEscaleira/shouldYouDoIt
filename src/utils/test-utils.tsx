import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { MemoryRouter as Router } from "react-router-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

import store, { initialStoreState } from "../store";

export const renderWithRedux = (ui: JSX.Element, initialState?: {}) => {
  const state = {
    ...initialStoreState,
    ...initialState
  };
  return {
    ...render(
      <Provider store={store(state)}>
        <Router>{ui}</Router>
      </Provider>
    )
  };
};

export const renderWithRouter = (ui: JSX.Element) => ({
  ...render(<Router>{ui}</Router>)
});

const middlewares = [thunk];
export const mockStore = configureStore(middlewares);
