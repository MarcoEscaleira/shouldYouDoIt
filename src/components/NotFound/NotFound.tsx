import React from "react";
import { Container, Label, Text, Button } from "./styled-components";

const NotFound: React.FC = () => (
  <Container>
    <Label>404</Label>
    <Text>Oops, we can't seem to find the page you're looking for.</Text>
    <Button to="/">Home</Button>
  </Container>
);

export default NotFound;
