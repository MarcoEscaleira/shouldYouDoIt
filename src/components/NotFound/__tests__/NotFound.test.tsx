import React from "react";
import NotFound from "../NotFound";
import { renderWithRedux } from "~/utils/test-utils";

describe("Not Found test suite", () => {
  it("Should render Not Found correctly", () => {
    const { container } = renderWithRedux(<NotFound />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
