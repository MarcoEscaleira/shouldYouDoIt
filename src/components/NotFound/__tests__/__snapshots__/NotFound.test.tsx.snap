// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Not Found test suite Should render Not Found correctly 1`] = `
<div
  class="sc-bdVaJa cCsnJS"
>
  <p
    class="sc-bwzfXH hZIatJ"
  >
    404
  </p>
  <p
    class="sc-htpNat sXacK"
  >
    Oops, we can't seem to find the page you're looking for.
  </p>
  <a
    class="sc-bxivhb diCfZq"
    href="/"
  >
    Home
  </a>
</div>
`;
